### DISCUSSION TITLE
Get request body

### DISCUSSION THREAD

**Discussion by Fnux8890** [elysiajs/elysia#528] [Q&A]
Hey. So in the following example form the docs on [guard](https://elysiajs.com/life-cycle/before-handle.html#guard)

In the example it has a get with a body. I'm not sure if the docs are wrong or if I'm missing something but I cant seem to get a body working with a get function.

```js
import { Elysia } from 'elysia'

new Elysia()
    .guard(
        {
            beforeHandle({ set, cookie: { session } }) {
                if (!validateSession(session.value))
                    return (set.status = 'Unauthorized')
            }
        },
        (app) =>
            app
                .get('/user/:id', ({ body }) => signUp(body))
                .post('/profile', ({ body }) => signIn(body), {
                    beforeHandle: isUserExists
                })
    )
    .get('/', () => 'hi')
    .listen(3000)
```

Here is my current code:

```js
import { logger } from '@grotto/logysia';
import { Elysia } from 'elysia'
import {helmet} from "elysia-helmet";

new Elysia()
    .use(helmet())
    .use(logger())
    .get('/', () => 'Landing')
    .get('/hello', () => 'Hi')
    .guard({
        beforeHandle({}){
            console.log("Before")
        }
    }, (app) => app.get('/protected/:id', ({}) => {
        return {
                message: 'Protected',
        }
    }))
    .listen(3000, () => console.log('Server started on http://localhost:3000'));
```

**Comment by binyamin** [comment]
There must probably a typo. Sending a body with a GET request is not usually a good idea - see [mdn](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET).

**Reply by mtt-artis** [reply]
Hi 👋

Elysia doesn t parse body on method get and head 

See

https://github.com/elysiajs/elysia/blob/main/src%2Fcompose.ts#L403-408

And then 

https://github.com/elysiajs/elysia/blob/main/src%2Fcompose.ts#L639
